iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -111,7 +111,17 @@ private static Table loadTable(Configuration conf, String tableIdentifier, Strin
 
     if (catalog.isPresent()) {
       Preconditions.checkArgument(tableIdentifier != null, "Table identifier not set");
-      return catalog.get().loadTable(TableIdentifier.parse(tableIdentifier));
+
+      if (catalog.get() instanceof AutoCloseable) {
```

Comment:
I guess Java 21 provides a smarter way to express this cast.
https://sonarcloud.io/project/issues?id=apache_hive&pullRequest=6131&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true